controllers/forum.server.controller.js

```diff
@@ -18,107 +20,118 @@ exports.postViews = function(req, res) {
-    Forum.searchPostById(req.params.id, function(result) {
-        if (result.err) {
-            // Return the error message with the error status
-            res.status(result.status).send(result.err);
-        } else {
-            // Return the forum post document object with 200 status
-            res.json({"forumPost": result});
-        }
-    });
+    const postID = req.params.id ? req.params.id : false;
+
+    if (isValidDocumentID(postID)) {
+        Forum.searchPostById(req.params.id, function(result) {
+            if (result.err) {
+                // Return the error message with the error status
+                res.status(result.status).send(result.err);
+            } else {
+                // Return the forum post document object with 200 status
+                res.json({"forumPost": result});
+            }
+        });
+    } else {
+        res.status(400).send("Bad request");
+    }
 }
 
 /**
  * Modifies the data of an existing forum post matching a given ID using HTTP request object data
- * @param req HTTP request object containing the forum post database document fields being updated
- * @param res HTTP request response object containing the updated forum post database document data
+ * @param req HTTP request object containing forum post fields being updated, user ID and auth token for verification
+ * @param res HTTP request response status code and updated forum post data in JSON format or error message
  */
 exports.postUpdateById = function(req, res) {
     const
-        reqParams = req.params,
+        postID = req.params.id ? req.params.id : false,
+        userID = req.body.userID ? req.body.userID : false,
+        authToken = req.get(configParams.get('authToken')),
         reqBody = req.body;
     let forumUpdateParams;
 
-    // Check that a valid ID exists in the request parameters
-    let isBadRequest = !isValidDocumentID(reqParams.id);
-
-    if (!isBadRequest) {
-        // Set fields for updating to an object with either passed values or false to declare them as invalid
-        forumUpdateParams = {
-            "communityID": reqBody.communityID && reqBody.communityID.length > 2 ? reqBody.communityID : false,
-            "title": reqBody.title && reqBody.title.length > 0 ? reqBody.title : false,
-            "bodyText": reqBody.text || false,
-            "upVotes": reqBody.upVotes ? parseInteger(reqBody.upVotes, 0) : false,
-            "downVotes": reqBody.downVotes ? parseInteger(reqBody.downVotes, 0) : false,
-            "attachments": reqBody.images || false
-        }
-
-        // Check that there is not zero valid fields being updated, removes invalid fields
-        isBadRequest = !isAnyFieldValid(forumUpdateParams);
+    // Set fields for updating to an object with either passed values or false to declare them as invalid
+    forumUpdateParams = {
+        "communityID": reqBody.communityID && reqBody.communityID.length > 2 ? reqBody.communityID : false,
```

Comment:
Couldn't you use the isValidDocumentID here?

Reply:
For communityID? There is not yet any requirement for this. I was thinking about it and assume it will be a task we crate for the next team

Reply:
Oh I thought by communityID you meant the _id given by mongoDB, so I thought it would be the same format and should work with the isValidDocumentID right? If not then it's fine, feel free to resolved this comment.

Reply:
The communityID has not been implemented yet. This is what is in the schema for it:
```
 // The ID for the community the blog is associates with
    communityID: {
        type: String,
        required: true,
        trim: true,
        minlength: 3
    },
```